tests/transformations/mapfusion_test.py

```diff
@@ -265,7 +428,13 @@ def fusion_with_nested_sdfg_0(A: dace.int32[10], B: dace.int32[10], C: dace.int3
             A[i] = tmp[i] * 2
     
     sdfg = fusion_with_nested_sdfg_0.to_sdfg(simplify=True)
-    sdfg.apply_transformations(MapFusion)
+
+    # Because the transformation refuses to fuse dynamic edges.
+    #  We have to eliminate them.
+    for state in sdfg.states():
+        for edge in state.edges():
+            edge.data.dynamic = False
```

Comment:
If you look at the SDFG, you see that the data connection is not really dynamic, but there will always be a write.
However, my transformation can not handle dynamic memlets (I forbid them because according to my understanding it does not make sense to include them).
The old transformation just went along with them, but I think this was just because it was not checked and it worked here because the memlet is not really dynamic.

However, as the name of the test is indicating, it is more about the nested SDFG, so I will rewrite the test to make it clear.

TODO(phimuell): Construct the graph in a way that there is no dynamic memlet.